src/Razor/src/Microsoft.AspNetCore.Razor.Tools/GenerateCommand.cs

```diff
@@ -169,6 +170,7 @@ protected override bool ValidateArguments()
             var engine = RazorProjectEngine.Create(configuration, compositeFileSystem, b =>
             {
                 b.Features.Add(new StaticTagHelperFeature() { TagHelpers = tagHelpers, });
+                b.Features.Add(new DefaultTypeNameFeature());
```

Comment:
This is required for generic components. I'm trying to port all of these features to the Components codebase, and this is where I realized stuff was broken :)